src/Adapter/MSTest.TestAdapter/Execution/TestMethodInfo.cs

```diff
@@ -467,65 +468,84 @@ private void RunTestCleanupMethod(object classInstance, TestResult result)
         }
         catch (Exception ex)
         {
-            var cleanupError = new StringBuilder();
-            var cleanupStackTrace = new StringBuilder();
-            if (result.TestFailureException is TestFailedException testFailureException)
-            {
-                if (!StringEx.IsNullOrEmpty(testFailureException.Message))
-                {
-                    cleanupError.Append(testFailureException.Message);
-                    cleanupError.AppendLine();
-                }
+            testCleanupException = ex;
+        }
 
-                if (!StringEx.IsNullOrEmpty(testFailureException.StackTraceInformation?.ErrorStackTrace))
-                {
-                    cleanupStackTrace.Append(testFailureException.StackTraceInformation.ErrorStackTrace);
-                    cleanupStackTrace.Append(Environment.NewLine);
-                    cleanupStackTrace.Append(Resource.UTA_CleanupStackTrace);
-                    cleanupStackTrace.Append(Environment.NewLine);
-                }
-            }
+        // If testCleanup was successful, then don't do anything
+        if (testCleanupException == null)
+        {
+            return;
+        }
 
-            Exception realException = ex.GetRealException();
-            string formattedExceptionMessage = realException.GetFormattedExceptionMessage();
+        if (testCleanupException is TestFailedException)
```

Comment:
we could add the comment also here to be aligned with the other place where we take this shortcut